docs/useCanAccessCallback.md

```diff
@@ -0,0 +1,61 @@
+---
+layout: default
+title: "useCanAccessCallback"
+---
+
+# `useCanAccessCallback`
+
+This hook returns a callback to call the `authProvider.canAccess()` method for a provided resource and action (and optionally a record).
+
+## Usage
+
+`useCanAccessCallback` returns an async function that takes an object `{ action, resource, record }` as argument. This function returns an object with the following properties:
+
+- `canAccess`: a boolean determining whether the user has access to the resource.
+- `error`: the error that might have occurred.
+
+```jsx
+import { useCanAccessCallback, useRecordContext, useDataProvider, Button } from 'react-admin';
+
+const ResetViewsButton = () => {
+    const record = useRecordContext();
+    const dataProvider = useDataProvider();
+    const checkAccess = useCanAccessCallback();
+    const resetViews = async () => {
+        const { canAccess, error } = await checkAccess({ action: 'resetViews', resource: 'users', record });
```

Comment:
I think you should catch errors using a try/catch block

Reply:
Didn't you suggest to throw in https://github.com/marmelab/react-admin/pull/10166/files#r1747220540?

Reply:
Sorry, I don't see the target of that link.